crates/lace-bin/src/typechecking/mod.rs

```diff
@@ -0,0 +1,32 @@
+use crate::name_resolution::NameResolutions;
+use crate::parser::ast::Ast;
+use crate::parser::span::Spans;
+use crate::source_file::SourceFile;
+use crate::typechecking::constraint::TypeConstraintGenerator;
+use crate::typechecking::context::TypeContext;
+use crate::typechecking::error::TypeError;
+use crate::typechecking::solver::SolvedTypes;
+use bumpalo::Bump;
+
+pub mod constraint;
+pub mod constraint_generation;
+pub mod constraint_metadata;
+pub mod context;
+pub mod error;
+pub mod solver;
+pub mod ty;
+
+pub fn typecheck<'a, 'newa>(
```

Comment:
I think `'a` and `'newa` could have more semantic names. (`'ast` & `'types` or something like that)